File: C_Good_Third_Question.py
from bisect import bisect_left
n,m = map(int,input().split())
num = list(map(int,input().split()))
num.sort()
for i in range(n):
    x = int(input())
    b = bisect_left(num,x)
    l = len(num) - b
    if l < len(num) and l > len(num)//2:
        print("YES")
    else:
        print("NO")
    

File: C_Good_Third_Question.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int main()
{
    int n,m;
    cin >> n >> m;
    vector<int> num(n);
    for(int i = 0; i < n; i++)
        cin >> num[i];
    sort(num.begin(), num.end());
    for(int i = 0; i < n; i++)
    {
        int x;
        cin >> x;
        int b = lower_bound(num.begin(), num.end(), x);
        cout << b << endl;
        int l = num.size() - b;
        if(l < num.size() && l >= num.size()/2 + 1)
            cout << "YES" << endl;
        else
            cout << "NO" << endl;
    }
    return 0;
}